### PR TITLE
LaEace Peyton code review

### PR DESCRIPTION
**Pull request by laeacepj**
Analysis of the program.
I found the program to be well put together and straightforward.

Was the program available in UC Github on time?
I was able to access the program on time with no issues.

Is the program documented/commented well enough for you to understand?
There were no misunderstandings I was able to understand.

Does the program compile?
The program does compile

Rationale behind your changes.
Mostly just minor changes to follow best practices.

Links to three specific commits that you made to your own group's GitHub repository after Sprint 1's deadline, and before Sprint 2's deadline.  

https://github.com/jakeFarrish/MedMapper/commit/43afa165d0d8376526360b49e6f8318c8ad55b4d

I had another commit but I can no longer find it, unsure of where it went.

Technical concepts:

I finally had to research the difference between a val and a const val, I had been putting this off.
I learned how to update a version in the build.gradle file.
I researched redundant qualifiers.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -50,7 +50,7 @@ android {
 dependencies {
 
     implementation 'androidx.core:core-ktx:1.7.0'
-    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.3.1'
+    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.6.1'
```
Comment (laeacepj) [inline app/build.gradle:53]: Updated android lifecycle version, this fixed multiple errors on the main activities file due to failed imports.

File: app/src/main/java/com/taskruler/MainActivity.kt [hunk 2]
```diff
@@ -60,7 +60,7 @@ fun Greeting(name: String) {
         TextField(
             value = activityName,
             onValueChange = { activityName = it },
-            label = { Text(stringResource(R.string.activtyName)) }
+            label = { Text(stringResource(R.string.activityName)) }
```
Comment (laeacepj) [inline app/src/main/java/com/taskruler/MainActivity.kt:63]: fixed a spelling mistake.

File: app/src/main/java/com/taskruler/RetrofitClientInstance.kt [hunk 3]
```diff
@@ -9,14 +9,14 @@ import retrofit2.converter.gson.GsonConverterFactory
  */
 object RetrofitClientInstance {
     private var retrofit: Retrofit? = null
-    private val BASE_URL = "https://wrytheruc.github.io/"
+    private const val BASE_URL = "https://wrytheruc.github.io/"
```
Comment (laeacepj) [inline app/src/main/java/com/taskruler/RetrofitClientInstance.kt:12]: turned the val to a const val to follow best practices.

cite: https://kotlinlang.org/docs/coding-conventions.html#overload-layout

File: app/src/main/java/com/taskruler/RetrofitClientInstance.kt [hunk 4]
```diff
@@ -9,14 +9,14 @@ import retrofit2.converter.gson.GsonConverterFactory
...
 
     val retrofitInstance : Retrofit?
         get() {
             // has this object been created yet?
             if (retrofit == null) {
                 // create it
-                retrofit = retrofit2.Retrofit.Builder()
+                retrofit = Retrofit.Builder()
```
Comment (laeacepj) [inline app/src/main/java/com/taskruler/RetrofitClientInstance.kt:19]: including "retrofit2" into this was a redundant qualifier name. removed it.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <resources>
     <string name="app_name">TaskRuler</string>
-    <string name="activtyName">Activity Name</string>
+    <string name="activityName">Activity Name</string>
```
Comment (laeacepj) [inline app/src/main/res/values/strings.xml:3]: Another spelling mistake to ensure it still worked after changing the previous one.

File: app/src/main/java/com/taskruler/MainActivity.kt [hunk 6]
```diff
@@ -19,7 +19,7 @@ import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class MainActivity : ComponentActivity() {
 
-    private val viewModel: MainViewModel by viewModel<MainViewModel>()
+    private val viewModel: MainViewModel by viewModel()
```
Comment (discospiff) [inline app/src/main/java/com/taskruler/MainActivity.kt:22]: This same suggestion appeared in a previously submitted code review.
When that occurs, only the first gets credit.

File: app/src/main/java/com/taskruler/MainActivity.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/taskruler/MainActivity.kt:63]: Good.

File: app/src/main/java/com/taskruler/MainActivity.kt [hunk 7]
```diff
@@ -54,11 +54,13 @@ fun Greeting(name: String) {
         Button(onClick = { /*TODO*/ })
         {Text(text = "Task Timed")}
 
+        Button(onClick = { /*TODO*/ })
+        {Text(text = "Start Timer")}
```
Comment (discospiff) [inline app/src/main/java/com/taskruler/MainActivity.kt:58]: Why is the button moved here?  Is this changing design, or does it have a significant impact on technical debt?

File: app/src/main/java/com/taskruler/MainViewModel.kt [hunk 8]
```diff
@@ -24,7 +24,7 @@ class MainViewModel(var taskService : ITaskService = TaskService()) : ViewModel(
      */
     fun getTasks() {
         viewModelScope.launch {
-            var innerTasks = taskService.getTasks()
+            val innerTasks = taskService.getTasks()
```
Comment (discospiff) [inline app/src/main/java/com/taskruler/MainViewModel.kt:27]: Good.

**Comment by discospiff** [comment]
Some tech debt reduction in this branch, so it's worth reviewing and considering merging.
I'm not sure why the button moved, though.  If the group wishes to merge this branch, they should review the new location of the button first.

File: app/src/main/java/com/taskruler/MainActivity.kt (hunk 6 above)
Comment (laeacepj) [reply app/src/main/java/com/taskruler/MainActivity.kt:22]: This was a change I made in the previous code review, I did it for the same group so I used the same branch, is it possible you're looking at the previous pull request? 3 of the changes listed were from code review 1 but only got responses today. I am unsure of why these are all listed together but they are outdated. Please let me know if there is something I can do to straighten this out.

**Comment by wrytherUC** [comment]
Aaron and myself have manually updated our code based on these suggestions. Closing PR

**Pull request closed without merging**